### PR TITLE
Rebase on fresh project and use custom tcp to register

### PR DESCRIPTION
**Pull request by rivernate**
Register tool with custom TCP
_Closes PLAT-2299_

### CODE REVIEW

File: app/helpers/registration_helper.rb [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        s.formats == [ToolProxy::TOOL_PROXY_FORMAT]
+      end
+
+      # Retrieve and return the endpoint of the ToolProxy.collection service
+      URI.parse(tp_service.endpoint) unless tp_service.blank?
+    end
+
+    private
+
+    def fetch_tool_consumer_profile
```
Comment (westonkd) [inline app/helpers/registration_helper.rb]: Create generic request

Comment (westonkd) [inline app/helpers/registration_helper.rb]: Working on this now

File: app/helpers/registration_helper.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
+module RegistrationHelper
+  # ToolConsumerProfile
+  #
+  # Helps with retrieving and using custom tool consumer profiles
+  # during registration.
+  class ToolConsumerProfile
```
Comment (rivernate) [inline app/helpers/registration_helper.rb]: I'm working on removing this

**Comment by rivernate** [comment]
I'm good with this being merged

**Pull request merged**